data_structures/src/chain.rs

```diff
@@ -2566,15 +2501,12 @@ impl DataRequestState {
         mut self,
         tally: TallyTransaction,
         block_hash_tally_tx: &Hash,
-    ) -> Result<DataRequestReport, failure::Error> {
+    ) -> Result<DataRequestInfo, failure::Error> {
         if let DataRequestStage::TALLY = self.stage {
             self.info.tally = Some(tally);
             self.info.block_hash_tally_tx = Some(*block_hash_tally_tx);
```

Comment:
`data_request_into.current_stage` should be set to `None` when the data request is finished, now it stays at "TALLY".

```suggestion
            self.info.block_hash_tally_tx = Some(*block_hash_tally_tx);
            self.info.stage = None;
```

Reply:
I assume you mean setting `self.info.current_stage = None`